File: src/main/java/dk/fitfit/doconelin/service/TagService.java
package dk.fitfit.doconelin.service;

import dk.fitfit.doconelin.domain.Tag;
import dk.fitfit.doconelin.repository.TagRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Set;

@Service
public class TagService implements TagServiceInterface {
	private final TagRepository tagRepository;

	public TagService(TagRepository tagRepository) {
		this.tagRepository = tagRepository;
	}

	@Override
	public Set<Tag> findTagsStartingWith(String name) {
		return tagRepository.findTagsByNameIgnoreCaseStartingWith(name);
	}

	@Override
	public Set<Tag> findTagsByRank() {
		return tagRepository.findTagsByRank();
	}

	@Override
	public Tag save(Tag tag) {
		return tagRepository.save(tag);
	}

	@Override
	public List<Tag> findAll() {
		return tagRepository.findAll();
	}

	@Override
	public Tag findOne(long id) {
		return tagRepository.findOne(id);
	}

	@Override
	public long getRank(long id) {
		return tagRepository.getRank(id);
	}
}

File: src/main/java/dk/fitfit/oneliner/service/TagService.kt
package dk.fitfit.oneliner.service

import dk.fitfit.oneliner.domain.Tag
import dk.fitfit.oneliner.repository.TagRepository
import org.springframework.stereotype.Service

@Service
class TagService(private val tagRepository: TagRepository) : TagServiceInterface {

    override fun findTagsStartingWith(name: String): Set<Tag> {
        return tagRepository.findTagsByNameIgnoreCaseStartingWith(name)
    }

/*
    override fun findTagsByRank(): Set<Tag> {
        return tagRepository.findTagsByRank()
    }
*/

    override fun save(tag: Tag): Tag {
        return tagRepository.save(tag)
    }

    override fun findAll(): List<Tag> {
        return tagRepository.findAll()
    }

    override fun getOne(id: Long): Tag {
        return tagRepository.getOne(id)
    }

    override fun getRank(id: Long): Long {
        return tagRepository.getRank(id)
    }
}
